src/AutoRest.CSharp/Common/Output/Models/MethodSignature.cs

```diff
@@ -7,5 +7,11 @@
 
 namespace AutoRest.CSharp.Output.Models
 {
-    internal record MethodSignature(string Name, string? Description, string Modifiers, CSharpType? ReturnType, FormattableString? ReturnDescription, Parameter[] Parameters) : MethodSignatureBase(Name, Description, Modifiers, Parameters);
+    internal record MethodSignature(string Name, string? Description, string Modifiers, CSharpType? ReturnType, FormattableString? ReturnDescription, Parameter[] Parameters, bool IsPageable) : MethodSignatureBase(Name, Description, Modifiers, Parameters)
+    {
+        public MethodSignature(string Name, string? Description, string Modifiers, CSharpType? ReturnType, FormattableString? ReturnDescription, Parameter[] Parameters)
+            :this(Name, Description, Modifiers, ReturnType, ReturnDescription, Parameters, false)
+        {
+        }
+    }
```

Comment:
Can we add an optional `IsPageable` parameter, instead of adding a new ctor?

Something like:
```C#
internal record MethodSignature(string Name, string? Description, string Modifiers, CSharpType? ReturnType, FormattableString? ReturnDescription, Parameter[] Parameters, bool IsPageable = false) : MethodSignatureBase(Name, Description, Modifiers, Parameters);
```